src/libraries/System.Text.Json/src/System/Text/Json/Writer/JsonWriterOptions.cs

```diff
@@ -135,11 +137,33 @@ public bool SkipValidation
             }
         }
 
+        /// <summary>
+        /// Gets or sets the new line string to use when <see cref="Indented"/> is <see langword="true"/>.
+        /// The default is the value of <see cref="Environment.NewLine"/>.
+        /// </summary>
+        /// <exception cref="ArgumentOutOfRangeException">
+        /// Thrown when the new line string is not <c>\n</c> or <c>\r\n</c>.
+        /// </exception>
+        public string NewLine
+        {
+            get => (_optionsMask & NewLineBit) != 0 ? s_alternateNewLine : Environment.NewLine;
+            set
+            {
+                JsonWriterHelper.ValidateNewLine(value);
+                if (value is not null && value != Environment.NewLine)
```

Comment:
Can `value` be null after validation is done?

Reply:
I added guards into the two callers to handle the null case.